src/ln/channel.rs

```diff
@@ -3053,11 +3054,16 @@ impl Channel {
 	/// self.remove_uncommitted_htlcs_and_mark_paused()'d
 	pub fn get_channel_reestablish(&self) -> msgs::ChannelReestablish {
 		assert_eq!(self.channel_state & ChannelState::PeerDisconnected as u32, ChannelState::PeerDisconnected as u32);
+		assert_ne!(self.cur_remote_commitment_transaction_number, INITIAL_COMMITMENT_NUMBER);
 		msgs::ChannelReestablish {
 			channel_id: self.channel_id(),
 			next_local_commitment_number: INITIAL_COMMITMENT_NUMBER - self.cur_local_commitment_transaction_number,
-			next_remote_commitment_number: INITIAL_COMMITMENT_NUMBER - self.cur_remote_commitment_transaction_number -
-				if self.channel_state & (ChannelState::FundingSent as u32 | ChannelState::OurFundingLocked as u32) == (ChannelState::FundingSent as u32) { 1 } else { 0 },
+			// We have to set next_remote_commitment_number to the next revoke_and_ack we expect to
```

Comment:
Hmm yes bolt is quite unclear on dissociation between next_local_commitment_number and next_remote_commitment_number. Maybe comment could be more meaningful : "next_local_commitment_number is future index of our revocation secret for our next local_commitment_tx, whereas next_remote_commitment_number is present index at which we expect remote node reveals secret for its actual remote commitment tx secret in its next revoke_and_ack" Don't know if it's clearer but at least I've tried :)